### DISCUSSION TITLE
W6100 support

### DISCUSSION THREAD

**Discussion by dirkdickler** [khoih-prog/EthernetWebServer#61] [General]
Hello,  will be any update for this chip :-)  ? 
Thank you

**Comment by khoih-prog** [comment]
Hi @dirkdickler 

As I don't have any W6100 now, I'll first search and buy at least one W6100 board. Then will see, develop and add support to it.

Is it **popular** now ? I haven't seen it mentioned / supported anywhere, in other Ethernet libraries (EThernet, Ethernet2, Ethernet3, EthernetLarge, etc.).

Depending on the availability and shipping time, I don't know when I'll have it to start.

It's backorder now in [**mouser**](https://www.mouser.ca/ProductDetail/WIZnet/W6100-Ethernet-Shield?qs=PzGy0jfpSMsS5D%252BbH7tyug%3D%3D)

![image](https://user-images.githubusercontent.com/57012152/184552213-9bb5fd87-3684-4e48-94e4-5a3251b44a9e.png)

![image](https://user-images.githubusercontent.com/57012152/184552235-6f4af4a9-0493-4ea1-8f9b-9abca75e448c.png)

**Comment by dirkdickler** [comment]
Hello, I have my own board with W5100S, W6100   ->  SPI output.  I can send
you it if you want.
Thank you.

ne 14. 8. 2022 v 21:39 odesílatel Khoi Hoang ***@***.***>
napsal:

**Comment by khoih-prog** [comment]
Hi @dirkdickler 

I just added the support to W6100 IPv4 to [**Ethernet_Generic** library](https://github.com/khoih-prog/Ethernet_Generic) from new [**v2.8.0**](https://github.com/khoih-prog/Ethernet_Generic/releases/tag/v2.8.0). Support to IPv6 will be added in the future.

Therefore, this library will automatically support it by using

```cpp
#define USE_ETHERNET_GENERIC  true
```

If you're still interested and have time to help, please test and report any error / bug as I still haven't received the ordered W6100 boards until 2 weeks from now.

Thanks,

**Comment by khoih-prog** [comment]
FYI, the **W6100 IPv4** has been tested and so far OK, using [**Ethernet_Generic**](https://github.com/khoih-prog/Ethernet_Generic) and this [**EthernetWebServer**](https://github.com/khoih-prog/EthernetWebServer)

**Comment by dirkdickler** [comment]
Hello Hello, I'm totally out of time with testing,  and I'm terribly sorry.
When I finish with my projects so immediately testing your code, if I wil
find some bug I let you know.
1000x thank you for your job.

Radek.

so 7. 1. 2023 v 4:39 odesílatel Khoi Hoang ***@***.***>
napsal:

**Comment by khoih-prog** [comment]
HI @dirkdickler 

FYI, just release new and better LwIP-based [**WebServer_ESP32_W6100**](https://github.com/khoih-prog/WebServer_ESP32_W6100) library for `ESP32+LwIP W6100`. 

In the near future, it'll be the base for many Async libraries for W6100 Ethernet, such as **AsyncWebServer_W6100**, **AsyncUDP_W6100**, etc.

Please give it a try if you have time.

BR,
